tools/sense_studio/templates/record_video.html

```diff
@@ -0,0 +1,116 @@
+{% extends 'skeleton.html' %}
+
+{% block title %} Record video {% endblock %}
+
+{% block main %}
+<div id="main" class="ui main text container">
+    <h1 class="ui header">
+        Video List
+        <div class="sub header">
+            <i class="folder open icon"></i> {{ path }}
+            <br>
+            <i class="eye icon"></i> {{ label }}
+            <br>
+            <i class="database icon"></i> {{ split }}
+        </div>
+    </h1>
+    <div>
+        <a>Pre recording duration</a>
+        <input name="duration_selection" type="text"  id="pre_recording_duration" value="3" />
+        <p/>
+        <a>Recording duration</a>
+        <input name="duration_selection" type="text"  id="recording_duration" value="5"/>
+        <p/>
+        <button onclick="record_video()">record</button>
+        <p/>
+        <div class="container" width="640" height="480" >
+        <video id="player" width="640" height="480" controls autoplay>
+            <source src="video/Motion.mp4" type="video/mp4" />
+        </video>
+        <div class="overlay">
+            <p id= "text_overlay"> </p>
+        </div>
+        </div>
+    </div>
+        <script>
+		  var player = document.getElementById('player');
+		  var timestamp = Date.now();
+		  var pre_recording = 3;
+		  var recording = 5;
+
+
+            function record_video() {
+                navigator.mediaDevices.getUserMedia({ audio: false, video: true })
+                  .then(handleSuccess)
+            }
+
+
+		  var handleSuccess = function(stream) {
+		    var canvas = document.querySelector("canvas");
+
+			// Optional frames per second argument.
+			// var stream = canvas.captureStream(30);
+			var recordedChunks = [];
+
+			console.log(stream);
+			var options = { mimeType: "video/webm; codecs=vp9" };
+
+			function handleDataAvailable(event) {
+			  console.log("data-available");
+			  if (event.data.size > 0) {
+				recordedChunks.push(event.data);
+				console.log(recordedChunks);
+				download();
+			  } else {
+				// ...
+			  }
+			}
+
+			// TODO: send this data to flask server
+			function download() {
+			  const url = "{{ url_for("save_video", path=path, label=label, split=split) }}".replace(/([^:]\/)\/+/g, "$1");
+			  var blob = new Blob(recordedChunks, {
+				type: "video/webm"
+			  });
+			    const formData = new FormData();
+                formData.append("video", blob);
+                fetch(url, {
+                  method: 'POST',
+                  body: formData,
+                }).then(res => {
+                  if(res.ok) alert('Video saved');
+                }).catch(err => {
+                  alert(err);
+                });
+			}
+
+			pre_recording = parseFloat(document.getElementById("pre_recording_duration").value)
+			recording = parseFloat(document.getElementById("recording_duration").value)
+
+            var text = "PRE RECORDING: " + pre_recording + "s";
+			document.getElementById("text_overlay").innerHTML = text;
```

Comment:
Is this text overlay supposed to show a countdown? For me it just shows a permanent `PRE RECORDING: 3s` message.
![image](https://user-images.githubusercontent.com/5707898/106747313-05608700-6624-11eb-9483-264d6ff5ee2c.png)

Reply:
No, it is not a countdown, just tell you how many time left.
I don t know how to do this countdown in js

Reply:
A countdown would be very helpful though :thinking: Currently it's hard to tell when the video starts and stops